[PHP 8.4] `str_getcsv`: Fix `$escape` parameter deprecation

Ayesh opened the pull request:
In PHP 8.4, not passing the `$escape` parameter to the `str_getcsv` function emits a deprecation error. This fixes the notice by passing an empty string which disables the buggy escaping mechanism.

 - [Initiating RFC](https://wiki.php.net/rfc/kill-csv-escaping)
 - [Deprecation RFC](https://wiki.php.net/rfc/deprecations_php_8_4#deprecate_proprietary_csv_escaping_mechanism)
 - [PHP 8.4: CSV: The `$escape` parameter must be provided](https://php.watch/versions/8.4/csv-functions-escape-parameter)


Before submitting pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [x] Any new functionality is covered by tests.

williamdes reviewed (approved):
Thank you!
we should send this into QA_5_2

Ayesh commented:
Thank you @williamdes, I will check the CI failure and rebase to QA 5.2 branch.

Ayesh committed:
[PHP 8.4] str_getcsv: Fix $escape parameter deprecation
In PHP 8.4, not passing the `$escape` parameter to the `str_getcsv`
function emits a deprecation error. This fixes the notice by passing
the identical default value for the `$escape` parameter.

 - [Initiating RFC](https://wiki.php.net/rfc/kill-csv-escaping)
 - [Deprecation RFC](https://wiki.php.net/rfc/deprecations_php_8_4#deprecate_proprietary_csv_escaping_mechanism)
 - [PHP 8.4: CSV: The `$escape` parameter must be provided](https://php.watch/versions/8.4/csv-functions-escape-parameter)

Signed-off-by: Ayesh Karunaratne <ayesh@aye.sh>

Ayesh commented:
Thank you for reviewing this @MauricioFauth and @williamdes . 
I rebased these changes against `QA_5_2` branch and force-pushed. We do indeed have tests for the workarounds we have in place for the `str_getcsv` buggy behavior tests passed locally. Thank you.

kamil-tekiela commented:
I don't know about this change. I wrote this code before I knew that this escaping mechanism is buggy. In fairness, it's all MySQL's fault. What you are proposing is just a temporary fix. If PHP goes ahead with the removal of this feature in PHP 9 then we will be left with no solution for this. We'd probably need a custom parser for this like we had before but less broken. Or we can refactor all of this code and heed my advise from the code comments:

```
        // This should really be delegated to MySQL but since we also want to HTML encode it,
        // it is easier this way.
        // It future replace str_getcsv with $dbi->fetchSingleRow('SELECT '.$expressionInBrackets[1]);
```

Also, I'm not too fond of the fact that this code is duplicated in JS.

Ayesh commented:
@kamil-tekiela yeah you are right this is just silencing the symptom. In fact, my first attempt at "fixing" this was to pass `escape: ''` to disable the escaping mechanism altogether, but it turned out that phpmyadmin is aware and accounting for this. Thank you.
